experiments/Schema/src/Query.jl

```diff
@@ -22,156 +22,246 @@ module QueryLib
   struct Query
     dom::Types
     codom::Types
-    dom_names::Array{String,1}
-    codom_names::Array{String,1}
-    query::String
-    type::String
+    dom_names::Array{Int,1}
+    codom_names::Array{Int,1}
+    tables::Array{String,1}
+    fields::Array{Tuple{Int,String},1}
+    connections::Array{Tuple{Int,Int},1}
   end
 
-  # Make every element of the array unique by appending ":X" with X as some digit
-  # This is necessary to prevent namespace collisions between columns
-  uniquify(a::Array{String,1}) = begin
+  function to_sql(q::Query)::String
+    # Convert index to table alias
+    ind_to_tab(i) = "t$i"
 
-    a_n = Array{String,1}()
-    # Fill a_n with unique values
-    for i in 1:length(a)
-      cur_val = a[i]
+    # Index to table name + alias
+    ind_to_alias(i) = "$(q.tables[i]) AS $(ind_to_tab(i))"
 
-      # Iterate the digit until the value is unique
-      while cur_val in a_n
-        if occursin(r"_\d*$", cur_val)
-          cur_val = replace(cur_val, r"_\d*$" => (c) -> "_"*string(parse(Int, c[2:end])+1))
-        else
-          cur_val = cur_val*"_0"
-        end
+    # Get field name from field index
+    field_to_s(i) = ind_to_tab(q.fields[i][1])*".$(q.fields[i][2])"
+
+    # Select statement
+    select = "SELECT "*join(field_to_s.(vcat(q.dom_names, q.codom_names)),", ")*"\n"
+
+    # From statement
+    from = "FROM "*join(ind_to_alias.(collect(1:length(q.tables))), ", ")*"\n"
+
+    # Conditions
+    s_conn = Array{String,1}()
+
+    for conn in q.connections
+      # Don't include a value if it's just a variable (shows a variable loop)
+      if q.fields[conn[1]][2] == "*"
+        continue
       end
-      append!(a_n, [cur_val])
+
+      append!(s_conn,[field_to_s(conn[1])*"="*field_to_s(conn[2])])
     end
-    return a_n
+    condition = "WHERE "*join(s_conn, " AND ")*";"
+    return select*from*condition
   end
 
-  @syntax FreeAbBiRelMeetJoin(ObExpr,HomExpr) AbelianBicategoryRelations begin
-    otimes(A::Ob, B::Ob) = associate_unit(new(A,B), munit)
-    otimes(f::Hom, g::Hom) = associate(new(f,g))
-    compose(f::Hom, g::Hom) = associate(new(f,g; strict=true))
-    dagger(f::Hom) = distribute_unary(distribute_dagger(involute(new(f))),
-                                      dagger, otimes)
+  function eval_variables!(q::Query)
+    changes = 1
+    while changes > 0
+      changes = 0
+      for conn in q.connections
```

Comment:
this should probably use the language of `e in edges(q.graph)` or something since this is a database related tool, `conn` and `connection` should be reserved for the connection to the database.